src/Microsoft.ML.Data/DataLoadSave/TransformerChain.cs

```diff
@@ -128,6 +128,25 @@ public DataViewSchema GetOutputSchema(DataViewSchema inputSchema)
             return s;
         }
 
+        [BestFriend]
+        internal TransformerChain<ITransformer> RewindToLastPredictionTransformer()
+        {
+            int lastPredictorIndex = 0;
+            for (int i = _transformers.Length; i > 0; i--)
+            {
+                var current = _transformers[i - 1];
+                if (current is IPredictionTransformer<IPredictorProducing<float>>
+                    || current is IPredictionTransformer<IPredictorProducing<VBuffer<float>>>)
+                {
+                    lastPredictorIndex = i;
+                    break;
+                }
+            }
+            Contracts.Assert(lastPredictorIndex != 0, "No predictor found in the model.");
```

Comment:
>Assert [](start = 22, length = 6)

If your intention is to throw if there is no predictor, then this needs to be a `Check` instead of `Assert`. Otherwise, callers of this method need to make sure there is a predictor in the chain before calling it.